### PR TITLE
Mobile + Layout Changes

### PR DESCRIPTION
**Pull request by notAreYouScared**
Server Overview - Desktop
image

Server Overview - Mobile
image

View Schedules - Mobile ( Before ) 
image
View Schedules - Mobile ( After ) 
image

Edit Schedules - Mobile ( Before )
image
Edit Schedules - Mobile ( After )
image

### CODE REVIEW

File: app/Filament/Server/Resources/FileResource/Pages/EditFiles.php [hunk 1]
```diff
@@ -65,6 +67,10 @@ public function form(Form $form): Form
                     ->default(function () {
                         $split = explode('.', $this->path);
 
+                        if (end($split) === 'yml') { //TODO: Temp fix for yaml...
+                            return 'yaml';
+                        }
+
```
Comment (RMartinOscar) [inline app/Filament/Server/Resources/FileResource/Pages/EditFiles.php:73]:
```suggestion
```

Comment (notAreYouScared) [reply app/Filament/Server/Resources/FileResource/Pages/EditFiles.php:73]: This is ment to be here.

**Comment by TekExplorer** [comment]
Consider having the line only consider the usage numbers, and not the tile for the section. This feels like the line is over-reaching.
image

File: app/Filament/Server/Resources/FileResource/Pages/EditFiles.php [hunk 2]
```diff
@@ -65,6 +67,10 @@ public function form(Form $form): Form
...
                         return end($split);
```
Comment (lancepioch) [inline app/Filament/Server/Resources/FileResource/Pages/EditFiles.php:74]:
```suggestion
                        $ext = pathinfo($this->path, PATHINFO_EXTENSION);

                        if ($ext === 'yml') {
                            return 'yaml';
                        }

                        return $ext;
```

File: app/Filament/App/Resources/ServerResource/Pages/ListServers.php [hunk 3]
```diff
@@ -77,48 +78,60 @@ private function uptime(Server $server): string
...
         $usedDisk = Arr::get($server->resources(), 'disk_bytes', 0);
 
-        $used = config('panel.use_binary_prefix')
+        return config('panel.use_binary_prefix')
             ? Number::format($usedDisk / 1024 / 1024 / 1024, maxPrecision: 2, locale: auth()->user()->language) .' GiB'
             : Number::format($usedDisk / 1000 / 1000 / 1000, maxPrecision: 2, locale: auth()->user()->language) . ' GB';
+    }
 
-        $total = config('panel.use_binary_prefix')
+    // @phpstan-ignore-next-line
```
Comment (lancepioch) [inline app/Filament/App/Resources/ServerResource/Pages/ListServers.php:126]: Nope.

```suggestion
```

File: app/Filament/App/Resources/ServerResource/Pages/ListServers.php [hunk 4]
```diff
@@ -77,48 +78,60 @@ private function uptime(Server $server): string
     // @phpstan-ignore-next-line
     private function cpu(Server $server): string
     {
-        $cpu = Number::format(Arr::get($server->resources(), 'cpu_absolute', 0), maxPrecision: 2, locale: auth()->user()->language) . '%';
-        $max = Number::format($server->cpu, locale: auth()->user()->language) . '%';
+        return Number::format(Arr::get($server->resources(), 'cpu_absolute', 0), maxPrecision: 2, locale: auth()->user()->language) . '%';
+    }
 
-        return $cpu . ($server->cpu > 0 ? ' Of ' . $max : '');
+    // @phpstan-ignore-next-line
```
Comment (lancepioch) [inline app/Filament/App/Resources/ServerResource/Pages/ListServers.php:84]: Sorry, can't let anymore of these in!

```suggestion
```

**Review by lancepioch — DISMISSED** [review]
Looks great so far, we'll need to work together to figure out the phpstan issues so we're not just adding `@phpstan-ignore`'s everywhere!

**Comment by notAreYouScared** [comment]
> Looks great so far, we'll need to work together to figure out the phpstan issues so we're not just adding `@phpstan-ignore`'s everywhere!

If phpstan wasn't retarded and tagging functions as not used, they wouldn't be required. 

Use this branch, remove all the ignores and run php stan...

**Comment by notAreYouScared** [comment]
@lancepioch Your changes broke displaying the limit, if it's not unlimited, it just shows 0...

Also removed the functionally of `GiB` vs `GB` both options show `GB`.

File: app/Models/Server.php [hunk 5]
```diff
@@ -453,6 +455,49 @@ public function resources(): array
...
+        if ($amount === 0 & $limit) {
+            return 'Unlimited';
+        }
+
+        if ($percentage) {
+            $unit = '%';
+        }
+
+        return Number::format($amount / $bits ** $precision, maxPrecision: 2, locale: auth()->user()->language) . $unit;
+    }
```
Comment (Boy132) [inline app/Models/Server.php:499]: There is already a [`convert_bytes_to_readable` helper function](https://github.com/pelican-dev/panel/blob/main/app/helpers.php#L22) btw.

Also, this seems to be unhandy/ confusing for percentage values.

Comment (lancepioch) [reply app/Models/Server.php]: Thanks, I've reused that method!

File: resources/views/tables/columns/server-entry-column.blade.php [hunk 6]
```diff
@@ -1,65 +1,62 @@
+@php /** @var \App\Models\Server $server */ $server = $getRecord(); @endphp
```
Comment (Boy132) [inline resources/views/tables/columns/server-entry-column.blade.php:1]: Getting claustrophobic here.

```suggestion
@php
    /** @var \App\Models\Server $server */
    $server = $getRecord();
@endphp
```

**Review by Boy132 — CHANGES_REQUESTED** [review]

File: app/helpers.php [hunk 7]
```diff
@@ -28,10 +28,10 @@ function convert_bytes_to_readable(int $bytes, int $decimals = 2): string
             return '0 ' . $suffix[0];
         }
 
-        $base = log($bytes) / log($conversionUnit);
-        $f_base = floor($base);
+        $fromBase = log($bytes) / log($conversionUnit);
+        $base ??= floor($fromBase);
 
-        return round(pow($conversionUnit, $base - $f_base), $decimals) . ' ' . $suffix[$f_base];
+        return round(pow($conversionUnit, $fromBase - $base), $decimals) . ' ' . $suffix[$base];
```
[the remark this answers is not in the document]
Comment (lancepioch) [reply app/helpers.php:34]: That can be a separate PR

File: app/Filament/App/Resources/ServerResource/Pages/ListServers.php [hunk 8]
```diff
@@ -31,12 +28,13 @@ public function table(Table $table): Table
             ->columns([
                 Stack::make([
                     ServerEntryColumn::make('server_entry')
+                        ->label(fn (Server $server) => $server->name)
```
[the remark this answers is not in the document]
Comment (lancepioch) [reply app/Filament/App/Resources/ServerResource/Pages/ListServers.php:31]: Doesn't look like it lol

**Review by lancepioch — APPROVED** [review]

**Review by Boy132 — APPROVED** [review]

**Pull request merged**